### DISCUSSION TITLE
pag中egl资源的初始化和去初始化

### DISCUSSION THREAD

**Discussion by tang-1210** [Tencent/libpag#2596] [Q&A]
请问pag接口调用，egl的初始化是在什么那个接口初始化，以及在什么地方去初始化，现在我是在异步线程池中 执行pag_player_->setProgress(frame_index_ * 1.0 / total_frame_num_);
                auto status = pag_player_->flush();guo

                auto data = new uint8_t[bytes_len_];
                pag_surface_->readPixels(pag::ColorType::BGRA_8888, pag::AlphaType::Premultiplied, data,
                               pag_file_->width() * 4); 最后销毁资源时报错EGLDevice::onMakeCurrent() failure result = 0 error= 12294， 我想问下是egl初始化和去初始化以及中间获取图像是不在一个线程中导致的吗

**Comment by kevingpqi123** [comment]
建议将背景补充全，如什么平台，通过什么样的方式实现什么功能，哪里不符合预期，现在不太清楚你遇到的问题是啥

**Comment by jinwuwu001** [comment]
该问题长时间未回复，先关闭，后续有问题可重新提
